lib/activerecord-multi-tenant/model_extensions.rb

```diff
@@ -21,9 +21,9 @@ def partition_key
           # Avoid primary_key errors when using composite primary keys (e.g. id, tenant_id)
           def primary_key
             return @primary_key if @primary_key
-            return @primary_key = super || DEFAULT_ID_FIELD if Rails::VERSION::MAJOR < 5
+            return @primary_key = super || DEFAULT_ID_FIELD if ActiveRecord::VERSION::MAJOR < 5
 
-            primary_object_keys = (connection.schema_cache.primary_keys(table_name) || []) - [partition_key]
+            primary_object_keys = ([*connection.schema_cache.primary_keys(table_name)] || []) - [partition_key]
```

Comment:
Thank you!  

Yes, Array.wrap would certainly work.  I was going to say that you might not want to because it's from ActiveSupport and would introduce a new dependency.  But then I realized you're already getting that when you use ActiveRecord.